wait/strategytarget_mock_test.go

```diff
@@ -0,0 +1,528 @@
+// Code generated by mockery. DO NOT EDIT.
```

Comment:
I'm not against having generated files with mockery, but I'd like to share may concerns.

We already have mocks in this package, but they are handcrafted for each wait strategy. Will the rest of wait strategies benefit from the mockery-generated mocks? If so, I'd migrate all the mocks to the mockery one (maybe in a follow-up PR?). Else, I'd stand with the manual stubs that we have, as they are simple to write and read.

Having mockery mocks would increase the cognitive load for contributing to the wait strategies, as it would mean having to install mockery and configure it right. Do you think that, in the case we move on with mockery mocks, we include a README in the wait package *and* update the wait docs in the site to inform developers on how to contribute to the wait strategies? That would reduce that load. Adding a make goal would be perfect too.

Reply:
SGTM, great to see alignment here 👏  If that's the plan, I think we can move on with this current state of the PR, and use it as starting point, maybe having a PR replacing old mocks, and another (or reusing that previous) one for the subpackage. Wdyt?

Reply:
Sounds good, [ticket raised](https://github.com/testcontainers/testcontainers-go/issues/2765) lets get this one merged @mdelapenya